std/assembly/math.ts

```diff
@@ -1437,7 +1437,8 @@ export namespace NativeMath {
   // @ts-ignore: decorator
   @inline
   export function round(x: f64): f64 {
-    return builtin_copysign<f64>(builtin_floor<f64>(x + 0.5), x);
+    let roundUp = builtin_ceil<f64>(x);
+    return select<f64>(roundUp, roundUp - 1.0, roundUp - 0.5 <= x);
```

Comment:
the `roundUp - 0.5 <= x` is a littler harder to think out than `roundUp - x <= 0.5` as the second one has no inexact operations